users/vieting/experiments/switchboard/ctc/feat/baseline_args.py

```diff
@@ -81,14 +81,18 @@ def get_nn_args_single(
     feature_net = feature_network_class(**feature_args).get_as_subnetwork()
     if preemphasis:
         for layer in feature_net["subnetwork"]:
-            if feature_net["subnetwork"][layer].get("from", "data") == "data":
-                feature_net["subnetwork"][layer]["from"] = "preemphasis"
+            layer_config = feature_net["subnetwork"][layer]
+            if layer_config.get("class") != "variable":
+                if feature_net["subnetwork"][layer].get("from") == "data":
+                    feature_net["subnetwork"][layer]["from"] = "preemphasis"
         feature_net["subnetwork"]["preemphasis"] = PreemphasisNetwork(alpha=preemphasis).get_as_subnetwork()
     if wave_norm:
         for layer in feature_net["subnetwork"]:
             if feature_net["subnetwork"][layer].get("from") == "data":
                 feature_net["subnetwork"][layer]["from"] = "wave_norm"
         feature_net["subnetwork"]["wave_norm"] = {"axes": "T", "class": "norm", "from": "data"}
+        if preemphasis:
+            feature_net["subnetwork"]["preemphasis"]["from"] = "wave_norm"
```

Comment:
Instead of adding very specific checks, I would prefer introducing a variable `source_layer` or so. Then you change the order of wave norm and preemphasis. Before the wave norm, you set it to `"data"` and for wave norm, you use that variable as `"from"` and then set it to `"wave_norm"`. Then for preemphasis, you again use that variable as `"from"` and then set it to `"preemphasis"`. That way, it would also be easier to extend then nested specific checks.